src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBadGenomicKmers.java

```diff
@@ -0,0 +1,135 @@
+    {
+        // Write all the reference sequences to a text file as a series of 10,000-base lines.
+        // For each contig, records are written with a K-1 base overlap
+        // (i.e., the last 62 bases in line n match the first 62 bases in line n+1) so that we don't miss any kmers
+        // due to line breaks -- we can just kmerize each record.
+        File refSequenceFile;
+        try
+        {
+            refSequenceFile = File.createTempFile("genomic_sequence","txt");
+            refSequenceFile.deleteOnExit();
+            final OutputStream os = new BufferedOutputStream(new FileOutputStream(refSequenceFile));
+
+            final SAMSequenceDictionary dict = ref.getReferenceSequenceDictionary(readsDict);
+            if ( dict == null )
+                throw new GATKException("No reference dictionary available.");
+
+            final int effectiveRecLen = REF_REC_LEN - KLEN + 1;
+            for ( final SAMSequenceRecord rec : dict.getSequences() )
+            {
+                final String seqName = rec.getSequenceName();
+                final int seqLen = rec.getSequenceLength();
+                final byte[] bases = ref.getReferenceBases(null,new SimpleInterval(seqName,1,seqLen)).getBases();
+                for ( int start = 0; start < seqLen; start += effectiveRecLen )
+                {
+                    os.write(bases,start,Math.min(REF_REC_LEN,seqLen-start));
+                    os.write('\n');
+                }
+            }
+
+            os.close();
+        }
+        catch ( IOException e )
+        { throw new GATKException("Unable to write reference text file for kmerizing.", e); }
+
+        // Turn that text file into an RDD, and do a classic map/reduce:
+        // Kmerize, mapping to a pair <kmer,1>, reduce by summing values by key, filter out <kmer,N> where N < 3,
+        // and collect the high frequency kmers back in the driver.
+        final int nParts = (int)(refSequenceFile.length()/CHUNK_SIZE + 1);
+        final JavaRDD<String> genomicBases = ctx.textFile(refSequenceFile.getPath(),nParts);
+        List<Kmer> killList =
+                genomicBases
+                        .flatMapToPair(seq ->
+                        {
+                            if ( seq.length() < KLEN )
+                                return Collections.emptyList();
+
+                            final List<Tuple2<Kmer,Integer>> kmers = new ArrayList<>(seq.length() - KLEN + 1);
+                            StringKmerizer itr = new StringKmerizer(seq,KLEN);
+                            while ( itr.hasNext() )
+                                kmers.add(new Tuple2<>(itr.next().canonical(KLEN),1));
+
+                            return kmers;
+                        })
+                        .reduceByKey(( v1, v2 ) -> v1 + v2)
+                        .flatMap(kv -> kv._2 > MAX_KMER_FREQ ? Collections.singletonList(kv._1) : Collections.emptyList())
```

Comment:
But doesn't do the same thing.  I'll replace with `.filter(kv->kv._2 > MAX_KMER_FREQ).map(kv->kv._1)`.